VocaDb.ResXFileCodeGenerator/StringBuilderGenerator.cs

```diff
@@ -35,17 +63,54 @@ public string Generate(Stream resxStream, GeneratorOptions options)
 
 		builder.Append("    ");
 		builder.Append(options.PublicClass ? "public" : "internal");
+		builder.Append(options.PartialClass ? " partial" : "");
 		builder.Append(options.StaticClass ? " static class " : " class ");
 		builder.AppendLine(options.ClassName);
 		builder.AppendLine("    {");
 
-		builder.Append("        private static ");
+		var indent = "        ";
+		if (options.InnerClassVisibility != InnerClassVisibility.NotGenerated)
+		{
+			builder.Append(indent);
+			builder.Append(options.InnerClassVisibility == InnerClassVisibility.SameAsOuter
+				? options.PublicClass ? "public" : "internal"
+				: options.InnerClassVisibility.ToString().ToLowerInvariant());
+			builder.Append(options.PartialClass ? " partial" : "");
+			builder.Append(options.StaticClass ? " static class " : " class ");
+
+			builder.AppendLine(string.IsNullOrEmpty(options.InnerClassName) ? "Resources" : options.InnerClassName);
+			builder.Append(indent);
+			builder.AppendLine("{");
+
+			indent += "    ";
+			if (!string.IsNullOrEmpty(options.InnerClassInstanceName))
+			{
+				if (options.StaticClass || options.StaticMembers)
+				{
+					reportError?.Invoke(Diagnostic.Create(s_memberWithStaticError,
+						Location.Create(options.FilePath!, new(), new())));
+				}
+
+				builder.Append(indent);
+				builder.Append("public ");
+				builder.Append(options.ClassName);
+				builder.Append(" ");
+				builder.Append(options.InnerClassInstanceName);
+				builder.Append(Constants.CultureInfoVariable);
```

Comment:
Is it intentional that `Constants.CultureInfoVariable` is appended here?

Reply:
Nope, good catch, and the unittest was wrong also.